libkbfs/cr_chains.go

```diff
@@ -954,9 +956,15 @@ func (ccs *crChains) copyOpAndRevertUnrefsToOriginals(currOp op) op {
 		newOp = realOp
 	}
 	for _, unref := range unrefs {
-		original, ok := ccs.originals[*unref]
-		if ok {
-			*unref = original
+		ok := true
+		// Loop over the originals, since if `changeOriginal` was
+		// called, there might be a path of them.
+		for ok {
+			var original BlockPointer
+			original, ok = ccs.originals[*unref]
```

Comment:
Ideally, `crChains.changeOriginal` would also update `originals` for every block pointing back to the one that's changing, but I didn't think introducing a reverse index or iterating through the entire list as worth it, so I just decided to follow the trail of breadcrumbs back to the new original instead.